Clean/GadgetsNew/Add8/Addition8.lean

```diff
@@ -19,35 +19,58 @@ open Provable (field field2 fields)
 open ByteLookup
 open Expression
 
-def add8 (input : Vector (Expression (F p)) 2) := do
-  let ⟨ [x, y], _ ⟩ := input
-  let z ← subcircuit Add8Full.circuit (vec [x, y, const 0])
+structure InputStruct (F : Type) where
+  x: F
+  y: F
+
+def Inputs (p : ℕ) : TypePair := ⟨
+  InputStruct (Expression (F p)),
+  InputStruct (F p)
+⟩
+
+instance : ProvableType (F p) (Inputs p) where
+  size := 2
+  to_vars s := vec [s.x, s.y]
+  from_vars v := ⟨ v.get ⟨ 0, by norm_num ⟩, v.get ⟨ 1, by norm_num ⟩ ⟩
+  to_values s := vec [s.x, s.y]
+  from_values v := ⟨ v.get ⟨ 0, by norm_num ⟩, v.get ⟨ 1, by norm_num ⟩ ⟩
+
+
+def add8 (input : (Inputs p).var) := do
+  let ⟨x, y⟩ := input
+  let z ← subcircuit Add8Full.circuit {
+    x := x,
+    y := y,
+    carry_in := const 0
+  }
```

Comment:
this can be made cleaner (for my aesthetics) by not repeating the names of variables that are called like the struct fields.

this suggestion applies across the board!

```suggestion
  let z ← subcircuit Add8Full.circuit { x, y, carry_in := const 0 }
```